homeworks/artem.bilenko_brilux/homework_6/homework_6_task_1.js

```diff
@@ -0,0 +1,37 @@
+function User() {
+    let fullName;
+    let phone;
+    const capitalLetter = new RegExp('\\b\\w');
+    const removeCharacters = new RegExp('[^\\d]', 'g');
+    const removeCharactersLeavingPlus = new RegExp('[^+\\d]', 'g');
+    Object.defineProperty(this, 'fullName', {
+        get() {
+            return fullName;
+        },
+        set(editedName) {
+            fullName = editedName
+                .toLowerCase()
+                .replace(capitalLetter, l => l.toUpperCase());
+        },
+    });
+    Object.defineProperty(this, 'phone', {
+        get() {
+            return phone;
+        },
+        set(editedPhone) {
+            if (editedPhone[0] !== '+') {
+                phone = editedPhone.replace(removeCharacters, '');
+            } else {
+                phone = editedPhone.replace(removeCharactersLeavingPlus, '');
+            }
```

Comment:
code is working, task is solved...

but FYI it could be much more optimized in terms of performance and readability...

1. `fullName` part looks ok
2. `phone` part could be optimized by splitting original string by non-digit chars, then adding optional `+` sign in front if it was a start char of original string, in your case you use RegExp for thing you actually don't need RegExp